subsys/net/lib/tftp/tftp_client.c

```diff
@@ -247,7 +308,87 @@ int tftp_get(struct sockaddr *server_addr, struct tftpc *client,
 		ret = TFTPC_REMOTE_ERROR;
 	}
 
-req_abort:
+get_abort:
+	k_mutex_unlock(&tftpc_lock);
+	close(sock);
+	return ret;
+}
+
+int tftp_put(struct sockaddr *server_addr, struct tftpc *client,
+	     const char *remote_file, const char *mode)
+{
+	int sock;
+	uint32_t tftpc_block_no = 1;
+	uint32_t tftpc_index = 0;
+	int ret;
+
+	if (client->user_buf == NULL || client->user_buf_size == 0) {
+		return -EFAULT;
+	}
+
+	sock = socket(server_addr->sa_family, SOCK_DGRAM, IPPROTO_UDP);
+	if (sock < 0) {
+		LOG_ERR("Failed to create UDP socket (IPv4): %d", errno);
```

Comment:
Why IPv4? The server address seems to be application-provided, so it could be IPv6 as well I guess?

Reply:
C&P from tftp_get(), updated both places